compiler-cli/src/dependencies.rs

```diff
@@ -543,6 +490,242 @@ fn same_requirements(
     Ok(left == right)
 }
 
+pub struct DependencyManagerConfig {
+    pub mode: Mode,
+    pub use_manifest: UseManifest,
+    pub check_major_versions: CheckMajorVersions,
+}
+
+impl Default for DependencyManagerConfig {
+    fn default() -> Self {
+        Self {
+            mode: Mode::Dev,
+            use_manifest: UseManifest::Yes,
+            check_major_versions: CheckMajorVersions::No,
+        }
+    }
+}
+
+impl DependencyManagerConfig {
+    pub fn into_dependency_manager<Telem: Telemetry, P: dependency::PackageFetcher>(
+        self,
+        runtime: tokio::runtime::Handle,
+        package_fetcher: P,
+        telemetry: Telem,
+    ) -> DependencyManager<Telem, P> {
+        DependencyManager {
+            runtime,
+            package_fetcher,
+            telemetry,
+
+            mode: self.mode,
+            use_manifest: self.use_manifest,
+            check_major_versions: self.check_major_versions,
+        }
+    }
+}
+
+pub struct DependencyManager<Telem: Telemetry, P: dependency::PackageFetcher> {
```

Comment:
Remove these constraints please 🙏 Constraints are only needed on the functions.

Reply:
Done